src/components/WearablePreview/WearablePreview.tsx

```diff
@@ -0,0 +1,371 @@
+    )
+    const showThumbnailBoundariesParam = safeEncodeParam(
+      "showThumbnailBoundaries",
+      showThumbnailBoundaries
+    )
+    const peerUrlParam = safeEncodeParam("peerUrl", peerUrl)
+    const nftServerUrlParam = safeEncodeParam("nftServerUrl", nftServerUrl)
+    const typeParam = safeEncodeParam("type", type)
+    const panningParam = safeEncodeParam("panning", panning)
+    const lockAlphaParam = safeEncodeParam("lockAlpha", lockAlpha)
+    const lockBetaParam = safeEncodeParam("lockBeta", lockBeta)
+    const lockRadiusParam = safeEncodeParam("lockRadius", lockRadius)
+    const envParam = safeEncodeParam("env", dev ? "dev" : undefined)
+    const unityParam = safeEncodeParam("unity", unity)
+    const unityModeParam = safeEncodeParam("mode", unityMode)
+
+    const url =
+      baseUrl +
+      "?" +
+      [
+        contractParam,
+        tokenParam,
+        itemParam,
+        profileParam,
+        urnParams,
+        urlsParams,
+        base64sParams,
+        skinParam,
+        hairParam,
+        eyesParam,
+        bodyShapeParam,
+        emoteParam,
+        cameraParam,
+        projectionParam,
+        zoomParam,
+        backgroundParam,
+        offsetXParam,
+        offsetYParam,
+        offsetZParam,
+        cameraXParam,
+        cameraYParam,
+        cameraZParam,
+        wheelZoomParam,
+        wheelPrecisionParam,
+        wheelStartParam,
+        disableBackgroundParam,
+        disableAutoRotateParam,
+        disableAutoCenterParam,
+        disableFaceParam,
+        disableDefaultWearablesParam,
+        disableDefaultEmotesParam,
+        disableFadeEffectParam,
+        showSceneBoundariesParam,
+        showThumbnailBoundariesParam,
+        peerUrlParam,
+        nftServerUrlParam,
+        typeParam,
+        panningParam,
+        lockAlphaParam,
+        lockBetaParam,
+        lockRadiusParam,
+        envParam,
+        unityParam,
+        unityModeParam,
+      ]
+        .filter((param) => !!param)
+        .join("&")
+    return url
+  }, [restProps, baseUrl, dev, unity])
+
+  const getOptions = useCallback(() => {
+    const options: PreviewOptions = {}
+
+    for (const key in restProps) {
+      if (typeof restProps[key] !== "function") {
+        options[key] = restProps[key]
+      }
+    }
+
+    return options
+  }, [restProps])
+
+  const sendUpdate = useCallback(
+    (options: PreviewOptions) => {
+      const { lastOptions } = state
+      // only send update if new options are different
+      if (!lastOptions || !equal(options, lastOptions)) {
+        // send message to iframe
+        if (iframeRef.current?.contentWindow) {
+          sendMessage(
+            iframeRef.current.contentWindow,
+            PreviewMessageType.UPDATE,
+            {
+              options,
+            }
+          )
+        }
+        // callback
+        onUpdate(options)
+
+        // store options on state
+        setState((prev) => ({ ...prev, lastOptions: options }))
```

Comment:
I think this `setState` is not required, as each time the code executes the sendUpdate method, after that  it does a setState